src/Data/Decimal.purs

```diff
@@ -0,0 +1,190 @@
+module Data.Decimal
+  ( Decimal()
+  , decimal
+  , getMantissa
+  , getExponent
+  , fromInt
+  , prettyDecimal
+  ) where
+
+import Prelude
+import Data.Array as A
+import Data.Either as E
+import Data.Maybe as M
+import Data.Ord as O
+import Data.String as S
+import Control.Monad.Rec.Class as Rec
+
+import Test.StrongCheck as SC
+import Test.StrongCheck.Gen as Gen
+
+type DecimalR =
+  { mantissa :: Int
+  , exponent :: Int
+  }
+
+-- | An abstract type for exact computing with decimal numbers of finite expansion.
+newtype Decimal = Decimal DecimalR
+
+instance showDecimal :: Show Decimal where
+  show (Decimal d) =
```

Comment:
It would be `(123,-12)`; but sorry, this was debug code which I forgot to replace. I'll replace it with what you suggest.